src/query/storage/m3/consolidators/tag_dedupe_map.go

```diff
@@ -163,3 +163,56 @@ func (m *tagDedupeMap) doUpdate(
 
 	return nil
 }
+
+func combineIters(first, second encoding.SeriesIterator) (encoding.SeriesIteratorAccumulator, error) {
+	acc, ok := first.(encoding.SeriesIteratorAccumulator)
+	if !ok {
+		var err error
+		acc, err = encoding.NewSeriesIteratorAccumulator(first, encoding.SeriesAccumulatorOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if err := acc.Add(second); err != nil {
+		return nil, err
+	}
+
+	return acc, nil
+}
+
+func stitchIfNeeded(
+	existing multiResultSeries,
+	tags models.Tags,
+	iter encoding.SeriesIterator,
+	attrs storagemetadata.Attributes,
+) (multiResultSeries, bool, error) {
+	// Stitching based on matching start/end.
+	if iter.Start().Equal(existing.iter.End()) {
```

Comment:
Does `Start()` and `End()` on the iterators return the times which were used to fetch data from dbnodes, or the times of earliest and latest datapoints returned? I'm wondering what would happen if a timeseries would temporarily disappear (e.g. due to instance being down or failed scrapes) and sticking point would fall into the gap. If `Start()` / `End()` tells the times of datapoints, then those would correspond to the ends of the gap and would not be equal.

Reply:
These are Start/End of the actual queries used to fetch the data. The fetch returns compressed blocks, so there can actually be datapoints even outside of the queried time range (those are filtered out somewhere in the underlying iterators).